### PR TITLE
support running gosec from a configurable list of subdirs

### PR DESCRIPTION
**Pull request by ghbren**
Support option to run gosec only from specified subdirs.  Example salus.yaml
```yaml
scanner_configs:
  Gosec:
    run_from_dirs:
      - app1
      - app2
      - app3
```

### CODE REVIEW

File: lib/salus/scanners/gosec.rb [hunk 1]
```diff
@@ -7,26 +7,81 @@
...
+          run_from_dir(dir)
+        end
+      end
+
+      # gosec runs from multiple dirs
+      # success only if none of the runs set @gosec_failed = true
+      return report_success if @gosec_failed == false
+
+      log(JSON.pretty_generate(@gosec_json)) if !@gosec_json.empty?
+      report_stderr(@gosec_stderr) if @gosec_stderr.empty?
```
Comment (joshua-ostrom) [inline lib/salus/scanners/gosec.rb:40]: should this if be an unless ?

File: lib/salus/scanners/gosec.rb [hunk 2]
```diff
@@ -7,26 +7,81 @@
...
+      report_stdout(@gosec_stdout) if @gosec_stdout.empty?
```
Comment (joshua-ostrom) [inline lib/salus/scanners/gosec.rb:41]: should this if be an unless?

File: lib/salus/scanners/gosec.rb [hunk 3]
```diff
@@ -7,26 +7,81 @@
...
+      # 'run_from_dirs' specifies a list of subdirs to run salus from
+      # if not specified, then run_from_dir will mimic the original gosec 'run' behavior
+      return run_from_dir if @config['run_from_dirs'].nil?
+
+      @gosec_failed = false
+      @gosec_stderr = ''   # combined stderr on all runs (one for each configured subdir)
+      @gosec_stdout = ''   # ...      stdout ...
+      @gosec_json = {}     # combined json result on all runs
+      run_from_dirs = @config['run_from_dirs'].sort
+      @config.delete('run_from_dirs')
```
Comment (joshua-ostrom) [inline lib/salus/scanners/gosec.rb:19]: Any need to restore this after the run_rom_dir blocks completes?  i.e. making this call impotent.

File: lib/salus/scanners/gosec.rb (hunk 2 above)
Comment (ghbren) [reply lib/salus/scanners/gosec.rb:41]: Great catch! Fixed.

File: lib/salus/scanners/gosec.rb (hunk 3 above)
Comment (ghbren) [reply lib/salus/scanners/gosec.rb]: Updated to restore the hash.

**Review by joshua-ostrom — APPROVED** [review]

**Pull request merged**